frontend/src/deviceDetailsPage/configuration/UploadConfigDialog.jsx

```diff
@@ -30,13 +30,16 @@ export default class UploadConfigDialog extends React.Component {
   handleUpload() {
     const { deviceId } = this.props;
     const { file } = this.state;
-    uploadConfiguration(deviceId, file).then(() => {
-      this.afterUpload();
-    });
-  }
-
-  afterUpload() {
-    this.dialogElement.current.closeDialog();
+    uploadConfiguration(deviceId, file)
+      .then(() => {
+        snackbar("success", "Successfully uploaded configuration");
+      })
+      .catch(() => {
+        snackbar("error", "Upload failed");
+      })
+      .finally(()=>{
```

Comment:
Linter being a big baby, but otherwise looks good!

Reply:
Another question I just thought of, does the snackbar display before the dialog closes since it's in a finally block?

Reply:
yes. but it will close right after